core/domain/param_domain_test.py

```diff
@@ -60,6 +60,26 @@ def test_param_change_validation(self):
             ):
             param_domain.ParamChange('abc', 'Copier', ['a', 'b']).validate()
 
+        # Raise an error because the param_change name is not a string.
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Expected param_change name to be a string, '
+            'received'):
+            param_domain.ParamChange(3, 'Copier', {}).validate()
+
+        # Raise an error because the arg names in customization_args are not
+        # strings.
+        with self.assertRaisesRegexp(
+            Exception, 'Invalid parameter change customization_arg name:'):
+            param_domain.ParamChange('abc', 'Copier', {1: '1'}).validate()
+
+        # Raise an error because generator id is invalid.
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'is not a valid generator '
+            'for exploration parameters. Valid generators must not require '
```

Comment:
I don't understand the error message here. What does it mean: "Valid generators must not requite any initialization arguments"?

(When testing, make sure to verify that things make intuitive sense.)

Reply:
I changed the message to Generator id must be a string which seems more appropriate